packages/api/lib/granules.js

```diff
@@ -86,48 +81,35 @@ async function moveGranuleFilesAndUpdateDatastore(params) {
     collectionPgModel = new CollectionPgModel(),
     filesPgModel = new FilePgModel(),
     dbClient = await getKnexClient(),
+    esClient = await Search.es(),
   } = params;
-  let postgresCumulusGranuleId;
-  let writeToPostgres = true;
-
-  try {
-    const { name, version } = deconstructCollectionId(apiGranule.collectionId);
-    postgresCumulusGranuleId = await granulePgModel.getRecordCumulusId(dbClient, {
-      granule_id: apiGranule.granuleId,
-      collection_cumulus_id: await collectionPgModel.getRecordCumulusId(
-        dbClient,
-        { name, version }
-      ),
-    });
-  } catch (error) {
-    // If the granule or associated record hasn't been migrated yet
-    // run the 'original' dynamo update
-    if (error.name !== 'RecordDoesNotExist') {
-      throw error;
-    }
-    log.info(`Granule ${JSON.stringify(apiGranule)} has not been migrated yet, updating DynamoDb records only`);
-    writeToPostgres = false;
-  }
 
+  const { name, version } = deconstructCollectionId(apiGranule.collectionId);
+  const postgresCumulusGranuleId = await granulePgModel.getRecordCumulusId(dbClient, {
+    granule_id: apiGranule.granuleId,
+    collection_cumulus_id: await collectionPgModel.getRecordCumulusId(
+      dbClient,
+      { name, version }
+    ),
+  });
   const updatedFiles = [];
   const moveFileParams = generateMoveFileParams(apiGranule.files, destinations);
   const moveFilePromises = moveFileParams.map(async (moveFileParam) => {
     const { file } = moveFileParam;
     try {
-      // Update the datastores, then move files
       await dbClient.transaction(async (trx) => {
         const updatedFile = await moveGranuleFile(
           moveFileParam,
           filesPgModel,
           trx,
           postgresCumulusGranuleId,
-          writeToPostgres
+          apiGranule.granuleId
         );
-        updatedFiles.push(renameProperty('name', 'fileName', updatedFile));
+        updatedFiles.push(updatedFile);
       });
-      // Add updated file to postgresDatabase
     } catch (error) {
-      updatedFiles.push({ bucket: file.bucket, key: file.key, fileName: getNameOfFile(file) });
+      // TODO update this to use OG file object.   Yuck.    This sucks
+      updatedFiles.push({ ...file, fileName: getNameOfFile(file) });
```

Comment:
The TODO should be removed (oops 😓 ) prior to the PR: b9c3e06f0

Reply:
re: the other comment, what's going on here is what the docstring says: 

```
* @returns {Promise<Object>} - Object containing an 'updated'
*  files object with current file key values and an error object containing a set of
*  Promise.allSettled errors
*/
```

And that's because we need to set the actual record to the *actual* state of the files - this method tries to move all possible files, then needs to update CMR with what moved, *then* handles the errors. 

https://github.com/nasa/cumulus/blob/jk/CUMULUS-2782-update-movegranules-to-write-to-es/packages/api/lib/granules.js#L172